pkg/api/controllers/transaction_controller.go

```diff
@@ -192,40 +193,49 @@ func (ctl *TransactionController) RevertTransaction(c *gin.Context) {
 		return
 	}
 
-	respondWithData[*core.Transaction](c, http.StatusOK, tx)
+	respondWithData[core.Transaction](c, http.StatusOK, tx)
 }
 
 func (ctl *TransactionController) PostTransactionMetadata(c *gin.Context) {
 	l, _ := c.Get("ledger")
 
 	var m core.Metadata
 	if err := c.ShouldBindJSON(&m); err != nil {
-		panic(err)
+		ResponseError(c, ledger.NewValidationError("invalid metadata format"))
+		return
 	}
 
 	txId, err := strconv.ParseUint(c.Param("txid"), 10, 64)
 	if err != nil {
-		c.AbortWithStatus(http.StatusBadRequest)
+		ResponseError(c, ledger.NewValidationError("invalid transaction ID"))
 		return
 	}
 
-	if err := l.(*ledger.Ledger).SaveMeta(c.Request.Context(), core.MetaTargetTypeTransaction, txId, m); err != nil {
+	_, err = l.(*ledger.Ledger).GetTransaction(c.Request.Context(), txId)
```

Comment:
It is a breaking change. We need to validate this.

Reply:
what are we doing here exactly ? i'm not sure i understand

Reply:
Before this changes, we can add a metadata on a transaction while it is not existings.
It is ok on accounts, but not on a transaction. It is more an unexpected behavior than a breaking changes finally.

Reply:
I agree :) It is clear now @jdupas22?